driver/driver.py

```diff
@@ -20,81 +9,266 @@
-# above. If you change the number of processors for
-# each MOM6 simulation, you will need to change this.
-MOM6_config = {
-    "SIM_DAYS": 1, # length of simlations 
-    "DOMAIN_LAYOUT": "32,36",
-    "MASKTABLE": "mask_table.242.32x36"
+from smartsim.database import Orchestrator
+from smartsim.log import log_to_file
+
+
+def create_mom_ensemble( experiment, args ):
+    """Creates the ensemble object that stores all configuration options
+    """
+
+    mom6_batch_args = {
+        "C":args.ensemble_node_features,
+        "exclusive": None
+    }
+
+    ensemble_batch_settings = experiment.create_batch_settings(
+        nodes=args.ensemble_size*args.nodes_per_member,
+        time="1:00:00",
+        batch_args = mom6_batch_args
+    )
+
+    mom6_run_args = {
+        "nodes"  : args.nodes_per_member,
+        "ntasks" : args.nodes_per_member*args.ppn_per_member,
+        "exclusive": None
+    }
+
+    mom6_run_settings = experiment.create_run_settings(
+        args.mom6_exe_path,
+        run_command = "srun",
+        run_args = mom6_run_args
+    )
+
+    mom_ensemble = experiment.create_ensemble(
+        "MOM",
+        batch_settings = ensemble_batch_settings,
+        run_settings   = mom6_run_settings,
+        replicas       = args.ensemble_size
+    )
+
+    mom_ensemble.attach_generator_files(
+        to_configure=glob("../MOM6_config/configurable_files/*"),
+        to_copy="../MOM6_config/OM4_025",
+        to_symlink="/lus/cls01029/shao/dev/gfdl/MOM6-examples/ice_ocean_SIS2/OM4_025/INPUT"
+    )
+
+    return mom_ensemble
+
+def configure_mom_ensemble( ensemble, args, colocated_orchestrator, db_nodes ):
+    """Configures all the members of the MOM6 ensemble with runtime configureable parameters
+    """
+
+    MOM6_config_options = {
+        "SIM_DAYS": 1, # length of simlations
+        "EKE_MODEL": args.eke_model_name,
+        "EKE_BACKEND": args.eke_backend,
+        "DOMAIN_LAYOUT": args.domain_layout,
+        "MASKTABLE": args.mask_table
     }
-for model in mom_ensemble:
-    model.params = MOM6_config
+    # For now the next entries have to be there to avoid the generator bug
+    MOM6_config_options.update( {
+        "SMARTREDIS_COLOCATED":"False",
+        "SMARTREDIS_COLOCATED_STRIDE":0,
+        "SMARTREDIS_CLUSTER": "False"
+    })
+
+    if colocated_orchestrator:
+        MOM6_config_options.update( {
+        "SMARTREDIS_COLOCATED":"True",
+        "SMARTREDIS_COLOCATED_STRIDE":18,
+        })
+    elif db_nodes >= 3:
+        MOM6_config_options.update( {'SMARTREDIS_CLUSTER':'True'} )
+
+    for model in ensemble:
+        model.params = MOM6_config_options
+        model.register_incoming_entity(model)
+
+    return ensemble
+
+def add_colocated_orchestrator( ensemble, args ):
+    for model in ensemble:
+        model.colocate_db(
+            args.orchestrator_port,
+            ifname=args.orchestrator_interface,
+            limit_app_cpus=False
```

Comment:
I see you aren't using db_cpus which means the database is only getting 1 thread/core each run. I would recommend adding that in and making it a configurable parameter.